app/controllers/twitter/callbacks_controller.rb

```diff
@@ -49,10 +49,18 @@ def create_inbox
       twitter_access_token_secret: parsed_body['oauth_token_secret'],
       profile_id: parsed_body['user_id']
     )
-    account.inboxes.create!(
+    inbox = account.inboxes.create!(
       name: parsed_body['screen_name'],
       channel: twitter_profile
     )
+    save_profile_image(inbox)
+    inbox
+  end
+
+  def save_profile_image(inbox)
+    response = twitter_client.user_show(screen_name: inbox.name)
```

Comment:
Should we rescue from failures here?

Reply:
This is already being handled in the transaction. The main function has exception handling.

Reply:
@tejaswinichile  lets add a rescue for the function or add a condition. We don't need to prevent the inbox creation in case some error occurs while fetching the picture

```suggestion
    response = twitter_client.user_show(screen_name: inbox.name)
    if response. success?
       parsed_user_profile = JSON.parse(response.read_body)
       ::Avatar::AvatarFromUrlJob.perform_later(inbox, parsed_user_profile['profile_image_url_https'])
    end
```